File: medis-ui/src/app/main/song/song.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';

import { Observable, Subject } from 'rxjs';

import { Song, SongSearch } from './song.model';

@Injectable({
  providedIn: 'root',
})
export class SongService {
  baseUrl = '/api/song';

  constructor(private http: HttpClient) {
    console.log('Song service created!');
  }

  getAll(): Observable<Song[]> {
    return this.http.get<Song[]>(this.baseUrl);
  }

  delete(song: Song): Observable<HttpResponse<any>> {
    return this.http.delete(this.baseUrl + '/' + song.id, { observe: 'response' });
  }

  create(song: Song): Observable<HttpResponse<Song>> {
    return this.http.post<Song>(this.baseUrl, song, { observe: 'response' });
  }

  update(song: Song): Observable<HttpResponse<Song>> {
    return this.http.put<Song>(this.baseUrl, song, { observe: 'response' });
  }

  search(search: SongSearch): Observable<HttpResponse<Song[]>> {
    return this.http.post<Song[]>(this.baseUrl + '/search', search, { observe: 'response' });
  }
}

File: medis-service/src/main/java/io/mersys/medis/service/SongService.java
package io.mersys.medis.service;

import java.util.List;
import java.util.Optional;

import io.mersys.medis.service.dto.SongDTO;
import io.mersys.medis.service.dto.SongSearchDTO;

public interface SongService {

	List<SongDTO> getAll();

	Optional<SongDTO> get(String id);

	SongDTO create(SongDTO dto);

	SongDTO update(SongDTO dto);

	void delete(String id);

	List<SongDTO> search(SongSearchDTO dto);
}
